src/jvm/main/org/jetbrains/kotlinx/lincheck/Utils.kt

```diff
@@ -282,17 +292,17 @@ private class CustomObjectInputStream(val loader: ClassLoader, inputStream: Inpu
  * threads that are related to the specified [runner].
  */
 internal fun collectThreadDump(runner: Runner) = Thread.getAllStackTraces().filter { (t, _) ->
-    t is FixedActiveThreadsExecutor.TestThread && t.runnerHash == runner.hashCode()
+    t is TestThread && t.runnerHash == runner.hashCode()
```

Comment:
Instead of using the `runnerHash`, you can check whether the thread is a current runner's thread implicitly.